src/lib/whenIdleOrHidden.ts

```diff
@@ -14,25 +14,19 @@
  * limitations under the License.
  */
 
-import {onHidden} from './onHidden.js';
-import {runOnce} from './runOnce.js';
-
 /**
  * Runs the passed callback during the next idle period, or immediately
  * if the browser's visibility state is (or becomes) hidden.
  */
-export const whenIdle = (cb: () => void): number => {
+export const whenIdleOrHidden = (cb: () => void) => {
   const rIC = globalThis.requestIdleCallback || setTimeout;
 
-  let handle = -1;
-  cb = runOnce(cb);
   // If the document is hidden, run the callback immediately, otherwise
   // race an idle callback with the next `visibilitychange` event.
   if (document.visibilityState === 'hidden') {
     cb();
   } else {
-    handle = rIC(cb);
-    onHidden(cb);
+    rIC(cb);
+    document.addEventListener('visibilitychange', cb, {once: true});
```

Comment:
【discuss】why the cb should be called two times？
- when it's hidden：call it once，immediately
- when it's visible：call it once with rIC and another time when change to hidden

before the change，it‘s wrapped with runOnce will only call once

Reply:
Thanks for catching this! I've created to https://github.com/GoogleChrome/web-vitals/pull/541 to address this problem.

Reply:
Yes. Very nice catch!